R/sensitivity-spider-plot.R

```diff
@@ -8,91 +8,111 @@
 #'   respectively. If `NULL`, all included paths and parameters present in the
 #'   supplied `SensitivityCalculation` object will be displayed in the
 #'   visualization.
-#' @param xAxisLog,yAxisLog Logical that decides whether to display the axis on
-#'   logarithmic scale.
-#' @param savePlots Logical that decides whether you wish to save created
-#'   plot(s). They are not saved by default. Note that if there are multiple
-#'   output paths in your model, there will be multiple plots that will be
-#'   saved.
-#' @param outputFolder A character string describing path folder where plots
-#'   need to be saved. The path must be relative to the current working
-#'   directory. By default (`""`), the plots will be saved in the current
-#'   working directory (which can be found using `getwd()` function). This
-#'   parameter is relevant only if `savePlots` is set to `TRUE`.
-#' @param width,height Plot size in inches. If not supplied, uses the size of
-#'   current graphics device.
-#' @inheritParams ggplot2::ggsave
+#' @param yAxisType Character string, either "percent" (percentage change) or
+#'   "absolute" (absolute values), for y-axis data normalization. Default is "percent".
+#' @param xAxisScale Character string, either "log" (logarithmic scale) or "lin"
+#'   (linear scale), to set the x-axis scale. Default is "log".
+#' @param yAxisScale Character string, either "log" or "lin", sets the y-axis scale
+#'   similarly to `xAxisScale`. Default is "lin".
+#' @param yAxisFacetScales Character string, either "fixed" or "free", determines
+#'   the scaling across y-axes of different facets. Default is "fixed".
+#' @param defaultPlotConfiguration An object of class `DefaultPlotConfiguration`
+#' used to customize plot aesthetics.
 #'
-#' @import ggplot2
+#'   Supported parameters include:
+#'   - `legendPosition`: Position of the legend on the plot.
+#'   - `legendTitle`: Title displayed for the legend.
+#'   - `linesAlpha`: Alpha transparency for line elements.
+#'   - `linesColor`: Color of the line elements.
+#'   - `linesSize`: Thickness of the line elements.
+#'   - `pointsShape`: Shape of the point elements.
+#'   - `pointsSize`: Size of the point elements.
+#'   - `subtitle`: Subtitle text for the plot.
+#'   - `title`: Main title text for the plot.
+#'   - `titleSize`: Font size of the plot title.
+#'   - `xAxisScale`: Scale type for the x-axis (`"log"` or `"lin"`).
```

Comment:
What happens if `xAxisScale` is defined in in functions signature? Which value wins?

Reply:
Currently, defaultPlotConfiguration$xAxisScale, if modified and therefore not `NULL`, would override the settings in the function signature in `.updatePlotConfiguration`. I see that this might be confusing. 
We should document that any modifications to `defaultPlotConfiguration` will take precedence over the plot specific settings. Or discuss a better implementation.